runtime/interpreter/interpreter_expression.go

```diff
@@ -1251,14 +1251,15 @@ func (interpreter *Interpreter) VisitReferenceExpression(referenceExpression *as
 
 	case *sema.ReferenceType:
 		// Case (3): target type is non-optional, actual value is optional.
-		// Unwrap the optional and add it to reference tracking.
+		// This path shouldn't be reachable. This is only a defensive step
+		// to ensure references are properly created/tracked.
 		if someValue, ok := result.(*SomeValue); ok {
 			locationRange := LocationRange{
 				Location:    interpreter.Location,
 				HasPosition: referenceExpression.Expression,
 			}
 			innerValue := someValue.InnerValue(interpreter, locationRange)
-			interpreter.maybeTrackReferencedResourceKindedValue(innerValue)
+			return NewEphemeralReferenceValue(interpreter, typ.Authorized, innerValue, typ.Type)
```

Comment:
Is this a bugfix? Previously, this case was falling through, still returning a reference to a Some (https://github.com/onflow/cadence/pull/2958/files#diff-484f7e68f1a1b9f96531632abacc84c131416a281cb29a502ac3b98c041ddb78L1265). Now a reference to the inner value is returned.

Reply:
Not really a bug fix. This path isn't reachable at runtime, was there only as a defensive check for proper resource tracking. This is still the case: By returning a reference (to the inner value), it makes sure the inner value's reference is tracked (in the previous revision as well, it was the inner value that was tracked). So this keeps the same old behaviour.